admin.go

```diff
@@ -322,12 +320,44 @@ func healthcheck(rw http.ResponseWriter, _ *http.Request) {
 		*resultList = append(*resultList, result)
 	}
 
+	queryParams := r.URL.Query()
+
+	if queryParams["json"] != nil {
+
+		type Result map[string]interface{}
+
+		response := make([]Result, len(*resultList))
+
+		for i, currentResult := range *resultList {
+			currentResultMap := make(Result)
+			currentResultMap["name"] = currentResult[0]
+			currentResultMap["message"] = currentResult[1]
+			currentResultMap["status"] = currentResult[2]
+			response[i] = currentResultMap
+		}
+
+		JSONResponse, err := json.Marshal(response)
+		if err != nil {
+			http.Error(rw, err.Error(), http.StatusInternalServerError)
+		} else {
+			execJSON(rw, JSONResponse)
+		}
+
+		return
```

Comment:
I think extracting this block as a new method looks better. And the `JSONResponse` which is start with upper case character need to be renamed.